common/util.go

```diff
@@ -364,11 +364,30 @@ func IsStickyWorkerUnavailable(err error) bool {
 
 // IsResourceExhausted checks if the error is a service busy error.
 func IsResourceExhausted(err error) bool {
-	switch err.(type) {
-	case *serviceerror.ResourceExhausted:
+	return IsErrorType[*serviceerror.ResourceExhausted](err)
+}
+
+// IsNamespaceHandoverErr checks if the error is namespace handover error.
+// NOTE: we can't directly check if err == ErrNamespaceHandover because the
+// check won't pass if the error is returned by an RPC call.
+// (status field in the ErrNamespaceHandover is empty and doesn't equal to the one
+// in the returned rpc error).
+func IsNamespaceHandoverErr(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if err.Error() == ErrNamespaceHandover.Error() {
 		return true
 	}
-	return false
+
+	return IsNamespaceHandoverErr(errors.Unwrap(err))
+}
+
+// IsErrorType checks if any error in the error chain matches error type T
+func IsErrorType[T error](err error) bool {
```

Comment:
What happens if someone types `common.IsErrorType[serviceerror.NotFound](err)`? (i.e. using the struct type instead of the pointer to struct)

I assume it will always return false. Is there any way to catch that bug? Or should we be really defensive and check for both `T` and `*T` in here?